### DISCUSSION TITLE
Is it possible to transform a PDF page, including all pdf element using a transformation matrix.

### DISCUSSION THREAD

**Discussion by gbaryiames** [pymupdf/PyMuPDF#3339] [Q&A] [answered]
I would like to preform the following on an existing pdf page.

1) Increase the page width. (I know how to this this step)
2) Then I would like to iterate through each pdf element on a page and apply a transformation matrix that will translate each element to right by the amount the page width was increased 
3) Save the document.

**Comment by JorjMcKie** [comment, marked as answer]
This should do the job:
```python
doc=fitz.open("v110-changes.pdf")
page=doc[0]
mb = page.mediabox
diff=105  # how much wider
mb
Rect(0.0, 0.0, 595.3200073242188, 841.9199829101562)
mb.x1 += diff # increase width
mb  # show
Rect(0.0, 0.0, 700.3200073242188, 841.9199829101562)
page.set_mediabox(mb)  # update MediaBox
m=fitz.Matrix(1,0,0,1,diff,0)  # make a shift matrix
s = "%g %g %g %g %g %g cm " % tuple(m)  # convert to PDF format 
fitz.TOOLS._insert_contents(page,s.encode(),overlay=False)  # prepend page's content
doc.ez_save("x.pdf")
```

**Reply by gbaryiames** [reply]
Worked Great! Thank you

**Comment by robinwatts** [comment]
> I would like to preform the following on an existing pdf page.
> 
> 1. Increase the page width. (I know how to this this step)
> 2. Then I would like to iterate through each pdf element on a page and apply a transformation matrix that will translate each element to right by the amount the page width was increased
> 3. Save the document.

Is that not equivalent to just moving the left hand edge of the page left? i.e. do the page expansion by reducing x0 rather than increasing x1? No need to fiddle with the operator stream at all.

**Reply by JorjMcKie** [reply]
@robinwatts - right: the x0 coordinate of the MediaBox could be set to `-diff` instead of the above approach.
